components/availability/src/Availability.svelte

```diff
@@ -171,20 +177,81 @@
     }
   };
 
+  // Consecutive same-availability periods of time, from earliest start_time to latest end_time.
+  const generateEpochs = (
+    slots: SelectableSlot[],
+    participant_threshold: number,
+  ) => {
+    let scale = scaleTime().domain([
+      slots[0].start_time,
+      slots[slots.length - 1].end_time,
+    ]);
+    let epochs = slots
+      .reduce((m, n) => {
+        if (
+          m[m.length - 1] &&
+          JSON.stringify(m[m.length - 1][0].available_calendars) ===
+            JSON.stringify(n.available_calendars)
+        ) {
+          m[m.length - 1].push(n);
+        } else {
+          m.push([n]);
+        }
+        return m;
+      }, [] as TimeSlot[][])
+      .map((epoch) => {
+        let status = "free";
+        if (
+          epoch[0].available_calendars.length === 0 ||
+          (epoch[0].available_calendars.length !== allCalendars.length &&
+            epoch[0].available_calendars.length <
+              allCalendars.length * (participant_threshold / 100))
+        ) {
+          status = "busy";
+        } else if (
+          epoch[0].available_calendars.length > 0 &&
+          epoch[0].available_calendars.length !== allCalendars.length
+        ) {
+          status = "partial";
+        }
+        return {
+          start_time: epoch[0].start_time,
+          offset: scale(epoch[0].start_time) * 100,
+          status,
+          height:
+            (scale(epoch[epoch.length - 1].end_time) -
+              scale(epoch[0].start_time)) *
+            100,
```

Comment:
`height` and `offset` because epochs, due to their variable length, don't play nice with css grid. They're set up as a `position: absolute` div and occupy the full height of it.